glow/event_log.go

```diff
@@ -0,0 +1,112 @@
+package glow
+
+// This file contains a simple event logger, designed to collect a limited
+// number of timestamped logs and dump them to file or terminal.
+
+import (
+	"container/list"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+)
+
+// Event.
+type event struct {
+	ts   time.Time
+	line string
+}
+
+// Options for the event log.
+type EventLogOptions struct {
+	Expiry     time.Duration // Older events will be removed. Default is no expiry.
+	LimitBytes int           // Limit the size of events to collect. Default is not to collect events.
+}
+
+// Event log.
```

Comment:
it's worth specifying in the docstring that the event log is fully in-memory and is implemented using a list.List

Reply:
Comments are a sort of sanity check on the code. It allows reviewers and other readers of the code to confirm that the things that they understand about the code match the things that the designer understands about the code.

It's especially helpful when using GPT to have comments that explain all of your ideas, because it makes it a lot easier for GPT to identify mistakes in the implementation, or at least highlight places where the code doesn't quite do what the comments aspire to.